File: arkts_static_test/03_Types/ets_cases/3.17.2_Type_Tuple/cross_lang_verify/runtime/TYP_03_17_02_010.swift
/**
 * Swift cross-language verification for TYP_03_17_02_010_RUNTIME_UNSAFE_GET
 */
import Foundation

let a: (String, String) = ("aa", "bb")
let t: Any = a

if let tuple = t as? (String, String) {
    let elem = tuple.1
    if elem != "bb" {
        fatalError("elem should be bb")
    }
}

print("verified")

File: arkts_static_test/03_Types/ets_cases/3.17.2_Type_Tuple/cross_lang_verify/runtime/TYP_03_17_02_010.java
/**
 * Java cross-language verification for TYP_03_17_02_010_RUNTIME_UNSAFE_GET
 */
class TYP_03_17_02_010 {
    public static void main(String[] args) {
        Object[] a = {"aa", "bb"};
        Object t = a;

        Object[] arr = (Object[]) t;
        Object elem = arr[1];
        if (!elem.equals("bb")) {
            throw new AssertionError("elem should be bb");
        }

        System.out.println("verified");
    }
}
